packages/react-router/src/link.tsx

```diff
@@ -320,35 +320,35 @@ export type LinkOptions<
 
 export type CheckPath<TRouteTree extends AnyRoute, TPass, TFrom, TTo> =
   ResolveRoute<TRouteTree, TFrom, TTo> extends never
-    ? CheckPathError<TRouteTree>
-    : TPass
+  ? CheckPathError<TRouteTree>
+  : TPass
 
 export type CheckPathError<TRouteTree extends AnyRoute> = {
   to: RoutePaths<TRouteTree>
 }
 
 export type ResolveRelativePath<TFrom, TTo = '.'> = TFrom extends string
   ? TTo extends string
-    ? TTo extends '.'
-      ? TFrom
-      : TTo extends `./`
-        ? Join<[TFrom, '/']>
-        : TTo extends `./${infer TRest}`
-          ? ResolveRelativePath<TFrom, TRest>
-          : TTo extends `/${infer TRest}`
-            ? TTo
-            : Split<TTo> extends ['..', ...infer ToRest]
-              ? Split<TFrom> extends [...infer FromRest, infer FromTail]
-                ? ToRest extends ['/']
-                  ? Join<['/', ...FromRest, '/']>
-                  : ResolveRelativePath<Join<FromRest>, Join<ToRest>>
-                : never
-              : Split<TTo> extends ['.', ...infer ToRest]
-                ? ToRest extends ['/']
-                  ? Join<[TFrom, '/']>
-                  : ResolveRelativePath<TFrom, Join<ToRest>>
-                : CleanPath<Join<['/', ...Split<TFrom>, ...Split<TTo>]>>
-    : never
+  ? TTo extends '.'
+  ? TFrom
+  : TTo extends `./`
+  ? Join<[TFrom, '/']>
+  : TTo extends `./${infer TRest}`
+  ? ResolveRelativePath<TFrom, TRest>
+  : TTo extends `/${infer TRest}`
+  ? TTo
+  : Split<TTo> extends ['..', ...infer ToRest]
+  ? Split<TFrom> extends [...infer FromRest, infer FromTail]
+  ? ToRest extends ['/']
+  ? Join<['/', ...FromRest, '/']>
+  : ResolveRelativePath<Join<FromRest>, Join<ToRest>>
+  : never
+  : Split<TTo> extends ['.', ...infer ToRest]
+  ? ToRest extends ['/']
+  ? Join<[TFrom, '/']>
+  : ResolveRelativePath<TFrom, Join<ToRest>>
+  : CleanPath<Join<['/', ...Split<TFrom>, ...Split<TTo>]>>
+  : never
```

Comment:
This shouldn't be getting flattened like this. Could you check if your prettier set-up is correctly picking up the prettier config at the root.

Reply:
hi @SeanCassiere thx for the feedback, have fixed the prettier issue